AK/StringUtils.cpp

```diff
@@ -63,7 +63,7 @@ bool matches(StringView str, StringView mask, CaseSensitivity case_sensitivity,
             break;
         case '\\':
             ++mask_ptr;
-            break;
+            [[fallthrough]];
         default:
             auto p = *mask_ptr;
```

Comment:
With this fall-through doesn't mean for a string like "foo\" (i.e. where `\` is the last character), it will do an out-of-bounds index?

Reply:
That is certainly the case.
The question is what to do when this happens. Ignore it by `break`ing like it was before, `return  false` to indicate an incomplete pattern, or just fall through without increasing `mask_ptr`?

Reply:
Hmm...how about treating it as a normal character? i.e. make `x\` match exactly `x\`.

Reply:
@alimpfard that's what I meant by "just fall through without increasing `mask_ptr`". Should be the case now.